api/v2/llm/suggestions.py

```diff
@@ -0,0 +1,111 @@
+from django.shortcuts import get_object_or_404
+from django_cas_ng.decorators import user_passes_test
+from ninja import Router
+from ninja.errors import HttpError
+
+from api.v2.llm.schema import ModifySuggestionSchema
+from common.ai_review.dto import SuggestionState
+from common.models import Comment, SuggestedComment
+from common.utils import is_teacher
+
+router = Router()
+
+
+@router.post(
+    "/rate/{suggestion_id}",
+    url_name="rate_submit_suggestion",
+    summary="Rate suggested comment",
+    description="Rate an AI-suggested comment as helpful or not.",
+)
+@user_passes_test(is_teacher)
+def rate_submit_suggestion(request, suggestion_id: int, query: dict):
+    suggestion = get_object_or_404(SuggestedComment, id=suggestion_id)
+
+    rating = query.get("rating")
+    if rating is None or not (0 <= rating <= 10):
+        raise HttpError(400, "Rating must be between 0 and 10")
+
+    suggestion.rating = rating
+    suggestion.save()
+
+    return {"status": "success"}
+
+
+@router.post(
+    "/resolve/{suggestion_id}",
+    url_name="accept_submit_suggestion",
+    summary="Accept suggested comment",
+    description="Accept an AI-suggested comment as a teacher.",
+)
+@user_passes_test(is_teacher)
+def accept_submit_suggestion(request, suggestion_id: int):
+    suggestion = get_object_or_404(SuggestedComment, id=suggestion_id)
+
+    created_comment = Comment.objects.create(
+        submit=suggestion.submit,
+        author=request.user,
+        text=suggestion.text,
+        source=suggestion.source,
+        line=suggestion.line,
+    )
+
+    suggestion.state = SuggestionState.ACCEPTED.value
+    suggestion.comment = created_comment
+    suggestion.save()
+
+    return created_comment.to_dto(can_edit=True, type="teacher", unread=False)
+
+
+@router.patch(
+    "/resolve/{suggestion_id}",
+    url_name="modify_submit_suggestion",
+    summary="Modify and accept suggested comment",
+    description="Modify a suggested comment text before accepting it.",
+)
+@user_passes_test(is_teacher)
+def modify_submit_suggestion(request, suggestion_id: int, body: ModifySuggestionSchema):
+    suggestion = get_object_or_404(SuggestedComment, id=suggestion_id)
+
+    created_comment = Comment.objects.create(
+        submit=suggestion.submit,
+        author=request.user,
+        text=body.modified_text if body.modified_text is not None else suggestion.text,
+        source=suggestion.source,
+        line=suggestion.line,
+    )
+
+    suggestion.state = SuggestionState.ACCEPTED.value
+    suggestion.comment = created_comment
+    suggestion.save()
+
+    return created_comment.to_dto(can_edit=True, type="teacher", unread=False)
+
+
+@router.delete(
+    "/resolve/{suggestion_id}",
+    url_name="decline_submit_suggestion",
+    summary="Reject suggested comment",
+    description="Reject an AI-suggested comment as a teacher.",
+)
+@user_passes_test(is_teacher)
```

Comment:
Note (not for this PR): we might want to think about more locked down permissions and don't allow any teacher to modify other teacher's suggestions.